src/utilities/core-utils.test.ts

```diff
@@ -91,6 +91,145 @@ describe("CoreUtils", () => {
+            expect(values).toHaveLength(2);
+        });
+
+        test("when given null enum, returns empty object", () => {
+            // Act
+            const result = CoreUtils.numericEnumToPojo(null);
+
+            // Assert
+            expect(result).toMatchObject({});
+            expect(result).toBeEmpty();
+        });
+    });
+
+    // #endregion numericEnumToPojo
+
+    // -------------------------------------------------------------------------------------------------
+    // #region ObjectToArray
+    // -------------------------------------------------------------------------------------------------
+
+    describe("objectToArray", () => {
+        test("convert object to array of values", () => {
+            // Arrange
+            const testObject = {
+                name: "Jane",
+                age:  30
+            }
+
+            // Act
+            const result = CoreUtils.objectToArray(testObject);
+
+            // Assert
+            expect(result).toContain(testObject.name);
+            expect(result).toContain(testObject.age);
+            expect(result).toHaveLength(2);
+        });
+
+
+        test("given an empty object, return an empty array", () => {
+            // Arrange
+            const testObject = {};
+
+            // Act
+            const result = CoreUtils.objectToArray(testObject);
+
+            // Assert
+            expect(result).toBeEmpty();
+        });
+    });
+
+    // #endregion ObjectToArray
+
+    // -------------------------------------------------------------------------------------------------
+    // #region sleep
+    // -------------------------------------------------------------------------------------------------
+
+    describe("sleep", () => {
+        test("schedule n milliseconds", () => {
+            // Arrange
+            jest.useFakeTimers();
+            const consoleSpy = jest.spyOn(console, "log");
+
+            // Act
+            CoreUtils.sleep(2000, true);
+
+            // At this point in time, only sleep start is printed out
+            expect(consoleSpy).toHaveBeenLastCalledWith("sleep start");
+
+            // Run 1 second later, "sleep end" has not been called yet
+            jest.advanceTimersByTime(1000);
+            expect(consoleSpy).not.toHaveBeenLastCalledWith("sleep end");
+
+            // Fast-forward until all timers have been executed
+            jest.runAllTimers();
+
+            // Assert
+            expect(consoleSpy).toHaveBeenLastCalledWith("sleep end");
+            jest.clearAllTimers();
+            jest.useRealTimers();
+        });
+    });
+
+    // #endregion sleep
+
+    // -------------------------------------------------------------------------------------------------
+    // #region sleepSync
+    // -------------------------------------------------------------------------------------------------
+
+    describe("sleepSync", () => {
+        test("delay n milliseconds", () => {
```

Comment:
Can this test name be written as more of a test case? Something along the lines of "given n milliseconds, it blocks execution until delay duration has completed"